classes/core/PKPQueueProvider.inc.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * @file classes/core/PKPQueueProvider.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPQueueProvider
+ * @ingroup core
+ *
+ * @brief Registers Events Service Provider and boots data on events and their listeners
+ */
+
+namespace PKP\core;
+
+use Illuminate\Queue\WorkerOptions;
+
+use PKP\config\Config;
+use PKP\Domains\Jobs\Job as PKPJobModel;
+
+class PKPQueueProvider
+{
+    public function runJobsAtShutdown(): void
+    {
+        $shouldRun = Config::getVar('queues', 'run_jobs_at_shutdown', false);
+
+        if (!$shouldRun) {
+            return;
+        }
+
+        $job = PKPJobModel
+            ::isavailable()
+                ->notexcedeedattempts()
+                ->limit(1)
+                ->first();
+
+        if (!$job) {
+            return;
+        }
+
+        // $artisan = app()->make('Illuminate\Support\Facades\Artisan');
```

Comment:
One of my first ideas, because at this point `app()` was already instantiated. But didn't work due to some notInstantiable errors.

I'm digging the rabbit hole further more.

Reply:
Likely also a namespacing issue. See https://github.com/Vitaliy-1/pkp-lib/commit/a8b4fcda4a094ccbef0d50fe4ec07c2720ac3d8e, haven't merged this yet